### ISSUE TITLE
False positive IDE0079

### ISSUE THREAD

**Issue by AbakumovAlexandr** [dotnet/roslyn#76379]
As requested in a [previous related issue](https://github.com/dotnet/roslyn/issues/51884#issuecomment-822136511), I'm opening another one for the same problem.

**Version Used**: 17.12.1

**Steps to Reproduce**:

1. Have a violation of some CA rule, for example, CA2000.
2. Suppress it in code with a `#pragma warning disable CA2000` \ `#pragma warning restore CA2000` pair:
```
#pragma warning disable CA2000
                var handler = new SocketsHttpHandler
                {
                    ...
                };
#pragma warning restore CA2000
```

**Diagnostic Id**: IDE0079: Remove unnecessary suppression

**Expected Behavior**: The suppress message is not marked as unnecessary.

**Actual Behavior**: The suppress message is marked as unnecessary but if removed, the build shows a warning that was suppressed.

**Comment by hugoqribeiro** [comment]
This is happening a lot on our code base. Very annoying.

Example:

```
public class Class1
{
    public void SimpleMethod()
    {
    }
}
```

_CA1822: Member 'SimpleMethod' does not access instance data and can be marked as static_

```
public class Class1
{
    [System.Diagnostics.CodeAnalysis.SuppressMessage("Performance", "CA1822:Mark members as static")]
    public void SimpleMethod()
    {
    }
}
```

_IDE0079: Remove unnecessary suppression_

Visual Studio 17.12.3
.NET SDK: 9.0.101

**Comment by deepakrathore33** [comment]
https://github.com/dotnet/roslyn/issues/75887
